src/index.ts

```diff
@@ -1,19 +1,70 @@
 import { Probot } from "probot";
 import fs from 'fs';
-const k8s = require('@kubernetes/client-node');
+import k8s, { V1Secret } from '@kubernetes/client-node';
+const { createAppAuth } = require("@octokit/auth-app");
+require('dotenv').config();
 
+
+const EXPIRATION_THRESHOLD = 5 * 60000;
+
+// K8s client for using k8s apis.
 const kc = new k8s.KubeConfig();
 kc.loadFromDefault();
 const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
 const k8sContext = kc.getCurrentContext();
-const namespace = (() => {
+const k8snamespace = (() => {
   if (k8sContext === 'inClusterContext') {
     return fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/namespace', 'utf8');
   } else {
     return k8sContext.split('/')[0];
   }
 })();
-console.log(`Operating in the namespace: ${namespace}`);
+console.info(`Operating in the namespace: ${k8snamespace}`);
+
+/**
+* Function that patches K8s secret with updated token and expiry time
+* @param    {Number} installId    Installation ID
+* @return   None
+*/
+async function checkExpiryUpdateToken(installId: Number){
+
+  const appSecret: any = await k8sApi.readNamespacedSecret("peribolos-"+installId, k8snamespace);
+
+  console.info(`Working on the Installation: ${appSecret.body.metadata.name}`);
+  const current_date = new Date();
+  const expiry_date = new Date(appSecret.body.metadata.annotations.expiresAt);
+
+  // check if token not expired
+  if (expiry_date.getTime() > (current_date.getTime() + EXPIRATION_THRESHOLD)){
+    console.info(`Token is still fresh for Installation: ${appSecret.body.metadata.name}`);
+    return
+  }
+
+  console.info(`Token is expired for Installation: ${appSecret.body.metadata.name}`);
+  const auth = await createAppAuth({ appId: process.env.APP_ID, privateKey: process.env.PRIVATE_KEY });
+  const resToken = await auth({ type: "installation", installationId: installId });
+
+
+  var secret: V1Secret = {
+    metadata: {
+      annotations: {
+        "expiresAt": resToken[1]
+      }
+    },
+    stringData: {
+      "token": resToken[0]
```

Comment:
As per your previous revision shouldn't this be?

```suggestion
      annotations: {
        "expiresAt": resToken.expiresAt
      }
    },
    stringData: {
      "token": resToken.token
```

Reply:
right, made the change :+1: